src/net.cpp

```diff
@@ -456,7 +471,14 @@ void CConnman::DumpBanlist()
 
 void CNode::CloseSocketDisconnect()
 {
-    Disconnect();
+    if (m_write_event) {
```

Comment:
I would prefer it if you could have some RAII wrappers around the libevent structures before adding all this.

Reply:
Agreed, see PR description :)